nupic/research/frameworks/pytorch/dataset_utils/dataset_utils.py

```diff
@@ -189,7 +191,13 @@ def split_dataset(dataset, groupby):
     _, indices = list(
         zip(*(sorted(list(indices_by_group.items()), key=lambda x: x[0])))
     )
-    return [Subset(dataset, indices=i) for i in indices]
+    subsets = [Subset(dataset, indices=i) for i in indices]
+
+    # Apply target_transform, if defined
+    if target_transform is not None:
```

Comment:
Why not apply the transforms to the dataset targets before splitting? 
Pytorch's torchvision datasets already support that, there is no need to do it only when spliting.
See `target_transform` argument in https://pytorch.org/docs/stable/torchvision/datasets.html#mnist

Reply:
the `split_dataset` function does the splitting based on the labels, so if we make all labels {0, 1} and then split, it will think there are only 2 classes.